src/month_delta.rs

```diff
@@ -0,0 +1,102 @@
+use chrono::{Datelike, NaiveDate};
+use polars::prelude::*;
+
+fn last_day_of_month(date: NaiveDate) -> NaiveDate {
+    if date.month() == 12 {
+        NaiveDate::from_ymd_opt(date.year() + 1, 1, 1)
+            .unwrap()
+            .pred_opt()
+            .unwrap()
+    } else {
+        date.with_day(1)
+            .unwrap()
+            .with_month(date.month() + 1)
+            .unwrap()
+            .pred_opt()
+            .unwrap()
+    }
+}
+
+pub(crate) fn impl_month_delta(start_dates: &Series, end_dates: &Series) -> PolarsResult<Series> {
+    if (start_dates.dtype() != &DataType::Date) || (end_dates.dtype() != &DataType::Date) {
+        polars_bail!(InvalidOperation: "polars_xdt.month_delta only works on Date type. Please cast to Date first.");
+    }
+    let start_dates = start_dates.date()?;
+    let end_dates = end_dates.date()?;
+
+    let month_diff: Int32Chunked = start_dates
+        .as_date_iter()
+        .zip(end_dates.as_date_iter())
+        .map(|(s_arr, e_arr)| {
+            s_arr.zip(e_arr).map(|(start_date, end_date)| {
+                let year_diff = end_date.year() - start_date.year();
```

Comment:
I've not looked at this in detail, but going to be honest, I'm a bit skeptical that such a complex function is correct..

Reply:
Ciao @MarcoGorelli, thanks for the review! 

I have to agree that the function is complex, what appears to be a simple operation required complexity to fulfil multiple case scenarios.

“The beauty of simplicity is the complexity it attracts.” - Tom Robbins

Worked on creating additional particular case scenarios and the function caught them all. 

```py
pl.DataFrame(
    {
        "start_date": [
            # One Month diff, end month has fewer days
            date(2023, 3, 31),  
            # Covering the leap year, 29th to 28th
            date(2020, 2, 29),  
            # Same as previous, reverse years
            date(2021, 2, 28),  
            # Same as previous, shift left
            date(2019, 2, 28),  
            # Border dates
            date(2024, 1, 31),  
            # Five years delta, checking for long distance
            date(2024, 1, 1),   
        ],
        "end_date": [
            date(2023, 4, 30),   
            date(2021, 2, 28),   
            date(2020, 2, 29),
            date(2020, 2, 29),   
            date(2024, 2, 4),    
            date(2029, 1, 1),  
        ],
    }
)
```

```
shape: (6, 4)
┌────────────┬────────────┬──────────────────────┬─────────────┐
│ start_date ┆ end_date   ┆ expected_month_delta ┆ month_delta │
│ ---        ┆ ---        ┆ ---                  ┆ ---         │
│ date       ┆ date       ┆ i64                  ┆ i32         │
╞════════════╪════════════╪══════════════════════╪═════════════╡
│ 2023-03-31 ┆ 2023-04-30 ┆ 1                    ┆ 1           │
│ 2020-02-29 ┆ 2021-02-28 ┆ 12                   ┆ 12          │
│ 2021-02-28 ┆ 2020-02-29 ┆ -12                  ┆ -12         │
│ 2019-02-28 ┆ 2020-02-29 ┆ 12                   ┆ 12          │
│ 2024-01-31 ┆ 2024-02-04 ┆ 0                    ┆ 0           │
│ 2024-01-01 ┆ 2029-01-01 ┆ 60                   ┆ 60          │
└────────────┴────────────┴──────────────────────┴─────────────┘
```

Reply:
I don't understand why you're using `last_day_of_month`, is it possible to do this a lot simpler? I think you just need to compare the month numbers, the day of the month, and the year?

Reply:
> I don't understand why you're using `last_day_of_month`, is it possible to do this a lot simpler? I think you just need to compare the month numbers, the day of the month, and the year?

Pairs of dates account for months with varying end days

1. **2019-02-28** to **2020-02-29**:
   - The first date, **2019-02-28**, is the last day of February in 2019.
   - The second date, **2020-02-29**, is the last day of February in 2020.
   - Although both dates represent the end of their respective years, they have different last days of the month:
     - In 2019, February has 28 days.
     - In 2020 (a leap year), February has 29 days.

2. **2023-03-31** to **2023-04-30**:
   - The first date, **2023-03-31**, is the last day of March in 2023.
   - The second date, **2023-04-30**, is the last day of April in 2023.
   - Again, both dates mark the end of their respective months, but March has 31 days, while April has 30 days.

I hope this clarifies the logic behind this approach for you! 😊

Reply:
thanks - ok to be fair
```python
In [46]: pl.Series([date(2022, 3, 31)]).dt.offset_by('1mo')
Out[46]:
shape: (1,)
Series: '' [date]
[
        2022-04-30
]
```
so this would be consistent with that

will try this out, would like to think there's a simpler way

Reply:
Precisely! 
I've been thinking about how to simplify it, for now sadly could not find a way. 
Sadly chrono, based on my limited research, doesn't have last day of month function.